src/dataflow/src/render/debezium.rs

```diff
@@ -118,6 +124,273 @@ where
     }
 }
 
+pub(crate) fn render_tx<G: Scope>(
+    data_id: GlobalId,
+    envelope: &DebeziumEnvelope,
+    input: &Stream<G, DecodeResult>,
+    tx_ok: Collection<G, Row, Diff>,
+    debug_name: String,
+) -> (
+    Stream<G, (Row, Timestamp, Diff)>,
+    Stream<G, (mz_dataflow_types::DataflowError, Timestamp, Diff)>,
+)
+where
+    G: ScopeParent<Timestamp = Timestamp>,
+{
+    let (before_idx, after_idx) = (envelope.before_idx, envelope.after_idx);
+    let hashed_id = data_id.hashed();
+
+    let (transaction_idx, tx_id_idx) = match envelope {
+        DebeziumEnvelope {
+            mode:
+                DebeziumMode::Full(DebeziumDedupProjection {
+                    transaction_idx,
+                    tx_id_idx,
+                    ..
+                }),
+            ..
+        }
+        | DebeziumEnvelope {
+            mode:
+                DebeziumMode::Ordered(DebeziumDedupProjection {
+                    transaction_idx,
+                    tx_id_idx,
+                    ..
+                }),
+            ..
+        }
+        | DebeziumEnvelope {
+            mode:
+                DebeziumMode::FullInRange {
+                    projection:
+                        DebeziumDedupProjection {
+                            transaction_idx,
+                            tx_id_idx,
+                            ..
+                        },
+                    ..
+                },
+            ..
+        } => (
+            *transaction_idx,
+            tx_id_idx.expect("render_tx should only be called when there's a transaction"),
+        ),
+        DebeziumEnvelope {
+            mode: DebeziumMode::None,
+            ..
+        } => panic!("render_tx should be called with a dedup mode"),
+    };
+
+    // XXX(chae): Should we try to avoid moving everything to one worker?
```

Comment:
exchanging based on transaction_id should work I think

Reply:
Hm, I'm not sure I follow. Why can't we write something like this? (with unwraps turned into some sensible default

``rust
Exchange::new(move |decode_result| {
    decode_result
        .value
        .unwrap()
        .iter()
        .nth(transaction_idx)
        .unwrap()
        .unwrap_list()
        .iter()
        .nth(tx_metadata_description.data_transaction_id_idx)
        .unwrap()
        .unwrap_str()
        .hashed()
)
```

Reply:
oh hmm good point. I think this should work (with, yeah, some defaults) because if, e.g. decode_result.value.is_none(), we don't have any hope of matching the transaction anyway so it doesn't matter which worker it ends up on!